lib/web/machineid.go

```diff
@@ -0,0 +1,70 @@
+// Teleport
+// Copyright (C) 2023  Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package web
+
+import (
+	"net/http"
+
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+
+	headerv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/header/v1"
+	machineidv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/machineid/v1"
+	"github.com/gravitational/teleport/lib/httplib"
+	"github.com/gravitational/teleport/lib/reversetunnelclient"
+)
+
+type CreateGitHubBotRequest struct {
```

Comment:
I'm a little unfamiliar with the Web API, but is there a reason we couldn't just make use of the entirety of *machineidv1.Bot here ?

Reply:
Some handlers define their custom request type, others just use the object's `Spec` field. Since we're reusing the bot spec fields here, I'll remove the new type and use the spec. Thanks.

Reply:
It looks like some handlers declare their custom request types, while other just use the object's `spec` field. 

Since we're accepting a field that is out of the spec (name), I think it is fine to use a custom request type here